results.tex

```diff
@@ -185,33 +190,22 @@ \subsubsection{Machine Learning algorithms}
 %Singular value decomposition (SVD) and principal component analysis (PCA)
 
 
-\paragraph{Algorithms to find best parameters}
-To find the parameters for these algorithms, several Machine Learning workhorse implementations can be used that can be re-used across different Machine Learning algorithm-families. These include:
+\paragraph{Parameter optimization}
+To find the best parameters for these algorithms, several Machine Learning workhorse implementations can be used that can be re-used across different Machine Learning algorithm-families. These include:
 \begin{itemize}
-	\item First-order techniques
-	\begin{itemize}
-		\item Stochastic gradient descent
-% 			Backpropagation
-		\item Stochastic dual coordinate ascent\cite{Shal13}
-		\item L-BFGS
-		\item Conjugate gradient
-	\end{itemize}
-	\item Second-order techniques
-	\begin{itemize}
-		\item Newton descent
-		\item Quasi-Newton descent
-	\end{itemize}
-	\item Coordinate descent
-	\item Markov-Chain Monte-Carlo
-	\item Variational inference
+	\item First-order techniques such as stochastic gradient descent, stochastic dual coordinate ascent\cite{Shal13}, L-BFGS or conjugate gradient
+	\item Second-order techniques such as Newton descent (which requires computing the Hessian which is generally infeasible) or Quasi-Newton descent (which approximates Newton descent)
+	\item Coordinate descent, which minimizes along coordinate directions
+	\item Markov-Chain Monte-Carlo, which samples from a probability distribution
+	\item Variational inference\cite{Blei17}, which approximates probability densities through optimization
 \end{itemize}
 
 
 % --------
 
 
 \subsubsection{Partitioning and distribution algorithms}
-This section contains an abstraction of the general design of a Distributed Machine Learning operating system that executes the Machine Learning workhorses across a wide variety of hardware.
+To execute the ML algorithms listed in the previous section of which the parameters are optimized also by the algorithms in the previous section, we can either execute them on a single computer (which is out of the scope of this paper) or in a distributed manner. There are several ways to partition the data and/or the program and to distribute these evenly across al machines.
```

Comment:
"previous section, of which"

Reply:
"are optimized also" -> "are also optimized", and perhaps a reference to optimization algorithms to make the sentence clearer.

Reply:
"al machines" -> "all machines"